qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/completions/QuteCompletionsForExpression.java

```diff
@@ -834,12 +836,33 @@ private void doCompleteExpressionForObjectPartWithParentNodes(Node part, Node no
 							}
 						}
 						break;
+					case WHEN:
+					case SWITCH:
+						// Completion for properties/methods of with object from #switch and #when
+						Parameter value = ((BaseWhenSection) section).getValueParameter();
+						if (value != null) {
+							String projectUri = template.getProjectUri();
+							ResolvedJavaTypeInfo whenSwitchJavaTypeInfo = javaCache.resolveJavaType(value, projectUri)
+									.getNow(null);
+							if (whenSwitchJavaTypeInfo != null) {

```

Comment:
The completion for fields must be available only for enum (we should manage that too for validation). To do that:

 * in JavaTypeKind (in ls and jdt project), add `Enum(4)`
 * in JDTTYpeUtils, create this method:

```java
public static JavaTypeKind getJavaTypeKind(IType type) throws JavaModelException {
		if (type.isClass()) {
			return JavaTypeKind.Class;
		}
		if (type.isEnum()) {
			return JavaTypeKind.Enum;
		}
		if (type.isInterface()) {
			return JavaTypeKind.Interface;
		}
		return JavaTypeKind.Unknown;
	}
```

 * replace https://github.com/redhat-developer/quarkus-ls/blob/c76ddfb1219cef1cb31206c023babfa888e60c76/qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/internal/template/JavaTypesSearch.java#L163 with 
 
`classInfo.setKind(JDTTYpeUtils.getJavaTypeKind(type);`

 * in https://github.com/redhat-developer/quarkus-ls/blob/c76ddfb1219cef1cb31206c023babfa888e60c76/qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/QuteSupportForTemplate.java#L438 add this line code:

`resolvedType.setKind(JDTTYpeUtils.getJavaTypeKind(type);`

At the end you can write:

`if (whenSwitchJavaTypeInfo != null && whenSwitchJavaTypeInfo.getJavaTypeKind() == JavaTypeKind.Enum) {`